src/stock-on-hand-mobile/pages/product-details.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+const ProductDetails = () => {
+
+    const history = useHistory();
+
+    const product = useSelector(state => state['productStockOnHand']['productData']);
+    const facility = useSelector(state => state['facilitiesStockOnHand']['facilityStockOnHand']);
+    const program = useSelector(state => state['programsStockOnHand']['programStockOnHand']);
+
+    const dateFormat = (date) => {
+        if (typeof date !== 'string'){
+            return (
+                date.getDate()+"/"+(date.getMonth()+1)+"/"+date.getFullYear()
+            )
+        }
+        else {
+            const correctDate = new Date(date);
+
+            return (
+                correctDate.getDate()+"/"+(correctDate.getMonth()+1)+"/"+correctDate.getFullYear()
+            )
+        }
+
+    }
+
+    const productData = [
+        { name: 'Stock on Hand', value: product?.stockOnHand },
+        { name: 'Product Code', value: product?.orderable?.productCode },
+        { name: 'Lot Code', value: product?.lot?.lotCode ?? 'No lot defined' },
+        { name: 'Expiry Date', value: product?.lot ? dateFormat(product?.lot?.expirationDate) : 'No lot defined'  },
```

Comment:
`        { name: 'Expiry Date', value: product?.lot ? dateFormat(product?.lot?.expirationDate) : 'No lot defined' },`
the ending has one redundant whitespace. Please remove to have like above.